swap/src/bin/swap.rs

```diff
@@ -327,6 +327,40 @@ async fn init_monero_wallet(
     Ok((monero_wallet, monero_wallet_rpc_process))
 }
 
+async fn qr_code(deposit_address: &bitcoin::Address) -> Result<String> {
```

Comment:
This doesn't need to be `async` because there is no `.await` in this function.

Reply:
You could generalize the function-signature to this if you are feeling fancy:

```suggestion
async fn qr_code(value: &impl ToString) -> Result<String> {
```

Reply:
Done.